lib/ts/fetch.ts

```diff
@@ -689,27 +699,29 @@ async function getFromCookies(name: string) {
     return undefined;
 }
 
-async function setTokenHeadersIfRequired(clonedHeaders: Headers, addRefreshToken: boolean = false) {
-    const transferMethod = AuthHttpRequest.config.tokenTransferMethod;
+async function setAuthorizationHeaderIfRequired(clonedHeaders: Headers, addRefreshToken: boolean = false) {
+    logDebugMessage("setTokenHeaders: adding existing tokens as header");
 
-    logDebugMessage("setTokenHeadersIfRequired: Adding st-auth-mode header: " + transferMethod);
-    clonedHeaders.set("st-auth-mode", transferMethod);
+    // We set the Authorization header even if the tokenTransferMethod preference set in the config is cookies
+    // since the active session may be using cookies. By default, we want to allow users to continue these sessions.
+    // The new session preference should be applied at the start of the next session, if the backend allows it.
 
-    logDebugMessage("setTokenHeaders: adding existing tokens as header");
+    const accessToken = await getTokenForHeaderAuth("access");
+    const refreshToken = await getTokenForHeaderAuth("refresh");
 
-    const token = await getTokenForHeaderAuth(addRefreshToken ? "refresh" : "access");
-    if (token) {
+    // We don't add the refresh token because that's only required by the refresh call which is done with fetch
+    // Still, we only add the Authorization header if both are present, because we are planning to add an option to expose the
+    // access token to the frontend while using cookie based auth
```

Comment:
```suggestion
    // access token to the frontend while using cookie based auth - so that users can get the access token to use 
    // in a (for example) websocket connection whilst using cookie based transfer method for http protocol.
    // In this case, the refresh token will be in cookies and not accessible to the frontend, and the access token
    // will also be in the cookies. So we don't want to auto add the access token as part of the authorization header
    // if the refresh token is not accessible to the frontend.
```